lib/turn_junebug_expressway/message_engine.ex

```diff
@@ -0,0 +1,68 @@
+defmodule TurnJunebugExpressway.MessageEngine do
+  use GenServer
+  use AMQP
+
+  require Logger
+  alias AMQP.Connection
+
+  alias TurnJunebugExpresswayWeb.Utils
+
+  @host Utils.get_env(:rabbitmq, :urn)
+  @exchange_name Utils.get_env(:rabbitmq, :exchange_name)
+  @queue_name Utils.get_env(:rabbitmq, :messages_queue)
+  @reconnect_interval 10_000
+
+  def start_link(opts \\ [name: __MODULE__]) do
+    GenServer.start_link(__MODULE__, nil, opts)
+  end
+
+  def init(_) do
+    send(self(), :connect)
+    {:ok, nil}
+  end
+
+  def publish_message(message) do
+    GenServer.call(__MODULE__, {:publish_message, message})
+  end
+
+  def get_connection do
+    GenServer.call(__MODULE__, :get)
+  end
+
+  def handle_call(:get, _, conn) do
+    {:reply, conn, conn}
+  end
+
+  def handle_call({:publish_message, message}, _from, conn) do
+    {:ok, channel} = AMQP.Channel.open(conn)
```

Comment:
We will also need to close this channel that we open.

Since we're not using multiple threads/processes here, we can probably put everything through a single channel. What about creating the channel on startup, and then using that channel to send all the messages? So on startup, instead of returning the connection, we return the channel. All channels are closed when the connection is closed, so we then don't need to worry about cleaning up channels

Reply:
Yes, let me do that. I wonder if that will solve my testing issue